run/src/main/scala/sbt/Fork.scala

```diff
@@ -45,13 +49,19 @@ final class Fork(val commandName: String, val runnerClass: Option[String]) {
       (classpathEnv map { value =>
         Fork.ClasspathEnvKey -> value
       })
-    val process = Process(command, workingDirectory, environment.toList: _*)
+    val jpb = new JProcessBuilder(command.toArray: _*)
+    workingDirectory foreach (jpb directory _)
+    environment foreach { case (k, v) => jpb.environment.put(k, v) }
+    if (connectInput)
+      jpb.redirectInput(Redirect.INHERIT)
+    val process = Process(jpb)
 
     outputStrategy.getOrElse(StdoutOutput) match {
-      case StdoutOutput        => process.run(connectInput)
-      case out: BufferedOutput => out.logger.buffer { process.run(out.logger, connectInput) }
-      case out: LoggedOutput   => process.run(out.logger, connectInput)
-      case out: CustomOutput   => (process #> out.output).run(connectInput)
+      case StdoutOutput => process.run(connectInput = false)
```

Comment:
I should note that my testing in the context of the Scala build was through this code path (`outputStrategy = StdoutOutput`). I _think_ the way the output is redirected should not have a bearing on how `connectInput` works, but if you can think of any other use cases to test for, I'd suggest doing some manual testing of this before the release.